util/webhook/webhook.go

```diff
@@ -44,429 +46,203 @@ const usernameRegex = `[a-zA-Z0-9_\.][a-zA-Z0-9_\.-]{0,30}[a-zA-Z0-9_\.\$-]?`
-	urlObj, err := url.Parse(webURL)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse repoURL '%s'", webURL)
-	}
+	regexpStr := fmt.Sprintf(regexpFormat, usernameRegex, regexEscapedHostname, regexEscapedPath)
 
-	regexEscapedHostname := regexp.QuoteMeta(urlObj.Hostname())
-	regexEscapedPath := regexp.QuoteMeta(urlObj.EscapedPath()[1:])
-	regexpStr := fmt.Sprintf(`(?i)^(http://|https://|%s@|ssh://(%s@)?)%s(:[0-9]+|)[:/]%s(\.git)?$`,
-		usernameRegex, usernameRegex, regexEscapedHostname, regexEscapedPath)
 	repoRegexp, err := regexp.Compile(regexpStr)
 	if err != nil {
-		return nil, fmt.Errorf("failed to compile regexp for repoURL '%s'", webURL)
+		return nil, fmt.Errorf("failed to compile regexp for repoURL '%s'", originalUrl)
 	}
 
 	return repoRegexp, nil
 }
 
-func (a *ArgoCDWebhookHandler) storePreviouslyCachedManifests(app *v1alpha1.Application, change changeInfo, trackingMethod string, appInstanceLabelKey string) error {
-	err := argo.ValidateDestination(context.Background(), &app.Spec.Destination, a.db)
-	if err != nil {
-		return fmt.Errorf("error validating destination: %w", err)
-	}
-
-	var clusterInfo v1alpha1.ClusterInfo
-	err = a.serverCache.GetClusterInfo(app.Spec.Destination.Server, &clusterInfo)
-	if err != nil {
-		return fmt.Errorf("error getting cluster info: %w", err)
-	}
-
-	var sources v1alpha1.ApplicationSources
-	if app.Spec.HasMultipleSources() {
-		sources = app.Spec.GetSources()
-	} else {
-		sources = append(sources, app.Spec.GetSource())
-	}
-
-	refSources, err := argo.GetRefSources(context.Background(), sources, app.Spec.Project, a.db.GetRepository, []string{}, false)
-	if err != nil {
-		return fmt.Errorf("error getting ref sources: %w", err)
-	}
-	source := app.Spec.GetSource()
-	cache.LogDebugManifestCacheKeyFields("moving manifests cache", "webhook app revision changed", change.shaBefore, &source, refSources, &clusterInfo, app.Spec.Destination.Namespace, trackingMethod, appInstanceLabelKey, app.Name, nil)
-
-	if err := a.repoCache.SetNewRevisionManifests(change.shaAfter, change.shaBefore, &source, refSources, &clusterInfo, app.Spec.Destination.Namespace, trackingMethod, appInstanceLabelKey, app.Name, nil); err != nil {
-		return fmt.Errorf("error setting new revision manifests: %w", err)
-	}
-
-	return nil
-}
-
-func sourceRevisionHasChanged(source v1alpha1.ApplicationSource, revision string, touchedHead bool) bool {
-	targetRev := parseRevision(source.TargetRevision)
-	if targetRev == "HEAD" || targetRev == "" { // revision is head
-		return touchedHead
-	}
-	targetRevisionHasPrefixList := []string{"refs/heads/", "refs/tags/"}
-	for _, prefix := range targetRevisionHasPrefixList {
-		if strings.HasPrefix(source.TargetRevision, prefix) {
-			return revision == targetRev
-		}
-	}
-
-	return source.TargetRevision == revision
+func GetWebUrlRegex(originalUrl string) (*regexp.Regexp, error) {
+	return getUrlRegex(originalUrl, `(?i)^((https?|ssh)://)?(%[1]s@)?((alt)?ssh\.)?%[2]s(:[0-9]+)?[:/]%[3]s$`)
```

Comment:
@robinlieb As far as I can tell, this regex covers the changes you introduced in #16292.

The updated regex matches the following sequence of parts:
1. optional: protocol (`http`, `https`, or `ssh`) followed by `://`
2. optional: username followed by `@`
   * _previously, the regex did not allow for usernames in `http`, `https`, or protocol-less URLs._
4. optional: `ssh` or `altssh` subdomain
   * _this is the tiny change I originally set out to introduce 😅_
5. required: hostname parsed from original URL
6. optional: `:` followed by port number
7. required: `:` or `/`
8. required: path parsed from original URL followed by optional `.git`

Feel free to suggest any additional test cases in `webhook_test.go` and I’ll incorporate them and make sure the regex works against them.